### ISSUE TITLE
The "already open" logic hard crashes Leo

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#199]
Leo opens a dialog when opening a file that may already be open in another copy of Leo.  The options are Yes, Yes to All, No and No to All.

For me, choosing No almost always hard-crashes python, giving the dreaded "Python has stopped working" message.  Other options also can crash Leo, iirc.

Imo, this is the worst bug in Leo.  Yes, the dialog is a useful feature, but it must not ever crash Leo.

This serious bug has been around far too long.

**Comment by tbnorth** [comment]
On Sat, 20 Jun 2015 06:40:14 -0700
"Edward K. Ream" <notifications@github.com> wrote:

> Leo opens a dialog when opening a file that may already be open in
> another copy of Leo.  The options are Yes, Yes to All, No and No to
> All.
> 
> For me, choosing No almost always hard-crashes python, giving the
> dreaded "Python has stopped working" message.  Other options also can
> crash Leo, iirc.

There was once a reproducible issue with Yes To All, and when I traced
it back it was nothing to do with the dialog code, which I think you
mistrust because of the way it handles remembering "to all" answers ;-)

As best I can remember, the the problem boiled down to given a list of
three outlines A, B, C to open, there were assumptions about global
stuff being initialized by the loading of A during the loading of B and
C which weren't true if the loading of A had been canceled.

The "to all" behavior of the dialog is very useful and I believe
innocent, but more importantly the underlying behavior of managing
multiple opening of the same outline is critical to avoid frustrating
data loss.

Cheers -Terry

**Comment by edreamleo** [comment]
@tbnorth > the problem boiled down to given a list of three outlines A, B, C to open, there were assumptions about global stuff being initialized by the loading of A during the loading of B and C which weren't true if the loading of A had been canceled.

Thanks for this.  Your analysis sounds right to me.

I'll discuss what seems to be a simpler alternative shortly in a new post.

**Comment by edreamleo** [comment]
I've just posted a detailed design for how to handle already-open files [here](https://groups.google.com/d/msg/leo-editor/jBdD4KlCrEM/V7OmTpM379kJ).

In brief, the proposal is to always open possibly-open files, followed by a warning dialog that demands no immediate action from the user.

**Comment by edreamleo** [comment]
Fixed at rev 5e4c4d3.  All tests pass.  From the checkin log:

```
Leo now warns in the log pane and raises a summary dialog about possibly already-open files.
Leo opens all files as usual, leaving it to the user to determine what to do.
    
A (new?) bug *won't* be fixed.  Closing a file removes the file's entry from g.app.db,
so reopening the file *again* won't give another warning.  This is a very minor matter.
```
